src/Swashbuckle.AspNetCore.SwaggerGen/Generator/SwaggerGenerator.cs

```diff
@@ -192,8 +192,13 @@ private Operation CreateOperation(ApiDescription apiDescription, ISchemaRegistry
             if (schema == null)
                 nonBodyParam.Type = "string";
             else
+            {
                 nonBodyParam.PopulateFrom(schema);
 
+                if (schema.Ref != null)
+                    nonBodyParam.Ref = schema.Ref;
```

Comment:
A very minor change, but could you move the following line into `SchemaExtension.PopulateFrom`:

```csharp
nonBodyParam.Ref = schema.Ref;
```

No need the for the `if` condition either